homeassistant/components/heos/__init__.py

```diff
@@ -51,6 +48,19 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass
+class HeosRuntimeData:
+    """Runtime data and coordinators for HEOS config entries."""
+
+    controller_manager: ControllerManager
+    group_manager: GroupManager
+    source_manager: SourceManager
+    players: dict[int, HeosPlayer]
+
+
+type HeosConfigEntry = ConfigEntry[HeosRuntimeData]
+
+
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
```

Comment:
Out of curiosity, is there a reason why there is a YAML setup for this? Can't we deprecate the YAML setup in favor of using the UI?

Reply:
Sure, it can be deprecated at this point. That's not related to this PR though, so happy to do that in a separate change.

Reply:
Oh definetly! I was mostly just surprised to see this code here haha